src/components/modals/DownloadParticipantsModal.tsx

```diff
@@ -54,83 +52,62 @@ export default function DownloadParticipantsModal({
-            ? {
-                key: 'project',
-                value: projectId.toString(),
-              }
-            : undefined,
+    try {
+      const participants = await querySubgraphExhaustive({
+        entity: 'participant',
+        keys: [
+          'wallet',
+          'totalPaid',
+          'balance',
+          'stakedBalance',
+          'unstakedBalance',
+        ],
+        orderBy: 'balance',
+        orderDirection: 'desc',
+        block: {
+          number: blockNumber,
         },
-        res => {
-          if (!res) return
-
-          res.participants.forEach(e => {
-            const p = parseParticipantJson(e)
-
-            let date = new Date((p.lastPaidTimestamp ?? 0) * 1000).toUTCString()
-
-            if (date.includes(',')) date = date.split(',')[1]
-
-            rows.push([
-              p.wallet ?? '--',
-              fromWad(p.balance),
-              fromWad(p.stakedBalance),
-              fromWad(p.unstakedBalance),
-              fromWad(p.totalPaid),
-              date,
-            ])
-          })
-
-          const expectNextPage =
-            res.participants.length && res.participants.length % pageSize === 0
-
-          if (expectNextPage) {
-            pageNumber++
-            query()
-          } else {
-            setLoading(false)
-
-            // Encode CSV content and download
-            const csvContent =
-              'data:text/csv;charset=utf-8,' +
-              rows.map(e => e.join(',')).join('\n')
-            const encodedUri = encodeURI(csvContent)
-            const link = document.createElement('a')
-            link.setAttribute('href', encodedUri)
-            link.setAttribute(
-              'download',
-              `@${handle}_holders-block${blockNumber}.csv`,
-            )
-            document.body.appendChild(link)
-
-            link.click()
-          }
+        where: {
+          key: 'project',
+          value: projectId.toString(),
         },
+      })
+
+      if (!participants) throw new Error('No data.')
+
+      participants.forEach(p => {
+        let date = new Date((p.lastPaidTimestamp ?? 0) * 1000).toUTCString()
+
+        if (date.includes(',')) date = date.split(',')[1]
+
+        rows.push([
+          p.wallet ?? '--',
+          fromWad(p.balance),
+          fromWad(p.stakedBalance),
+          fromWad(p.unstakedBalance),
+          fromWad(p.totalPaid),
+          date,
+        ])
+      })
+
+      const csvContent =
+        'data:text/csv;charset=utf-8,' + rows.map(e => e.join(',')).join('\n')
+      const encodedUri = encodeURI(csvContent)
+      const link = document.createElement('a')
+      link.setAttribute('href', encodedUri)
+      link.setAttribute(
+        'download',
+        `@${handle}_holders-block${blockNumber}.csv`,
       )
-    }
+      document.body.appendChild(link)
 
-    query()
+      link.click()
+
+      setLoading(false)
+    } catch (e) {
+      console.log('Error downloading participants', e)
```

Comment:
**suggestion**: what's your opinion on `console.error` instead?

Side note: it might be worth considering some exception tracking tool like Sentry going forward.

Reply:
i'm not sure why this error would ever actually get thrown, but down to use console.error

also down for considering sentry at some point